File: JavaScript/player.js
//player class
function Player (name, cards) {

	//private member variables
	var playerName = name; 
	var hand = cards;
	var sizeOfDeck = hand.length;
	
	//public method: returns the card at that pos or null if pos is invalid
	this.discardCard = function(pos) {
		if (pos < 0 || pos >= sizeOfDeck)
			return null
		var temp = hand[pos];
		sizeOfDeck--;
		for (var i = pos; i < sizeOfDeck; i++) {
			hand[i] = hand[i+1];
		}
	};
	
	//public method: adds a card to hand
	this.addCard = function(newCard) {
		hand[sizeOfDeck] = newCard;
		sizeOfDeck++;
	};
	
	//public method: returns the name
	this.getName = function() {
		return playerName;
	};
	
	//public method: prints the cards currently holding
	this.printHand = function() {
		for(var i = 0; i < sizeOfDeck; i++)
			console.log(hand[i].toString());	 
		console.log("Hand size is " + sizeOfDeck);
	};
}

File: src/Player.java
import java.util.ArrayList;


public class Player {

	private String name;
	private ArrayList <Card> hand;
	
	public Player(String x, ArrayList<Card> y) {
		name = x;
		hand = y;
	}

	public void printHand() {
		for (int i = 0; i < hand.size(); i++)
			System.out.print(hand.get(i).toString() + " ");
		System.out.println();		
	}
	
	//returns null if card selection is not valid, otherwise returns a card at location pos
	public Card discardCard(int pos) {
		if (pos >= hand.size() || pos < 0)
			return null;
		Card temp = hand.get(pos);
		hand.remove(pos);
		return temp;
	}
	
	//add a card to the array list
	public void addCard(Card newOne) {
		hand.add(newOne);
	}
	//gets the size of the players hand
	public int getHandSize() {
		return hand.size();
	}
	//get players name
	public String getName() {
		return name;
	}
}
